syntax = "proto3";

package quotesig;

// -----------------Cart service-----------------

service QuoteService {
    rpc OnNotifyTicks(Tick) returns (Empty) {}
    rpc OnNotifyOHLC(OHLC) returns (Empty) {}
    rpc OnNotifyTradeSignal(Signal) returns (Empty) {}
}

message Tick {
    string symbol = 1;
    string timestamp = 2;
    string best_bid_price = 3;
    string best_bid_amount = 4;
    string best_ask_price = 5;
    string best_ask_amount = 6;
}

message OHLC {
    string symbol = 1;
    string timestamp = 2;
    string open = 3;
    string high = 4;
    string low = 5;
    string close = 6;
}

message Signal {
    string symbol = 1;
    string timestamp = 2;
    int32 position = 3;
}

message Empty {}